sqlmesh/core/engine_adapter/base.py

```diff
@@ -2040,6 +2040,12 @@ def temp_table(
         Yields:
             The table expression
         """
+        if isinstance(name, str):
```

Comment:
you can just unconditionally do table = exp.to_table(name)

Reply:
you mean `name = exp.to_table(name)` right?
As we have `table = self._get_temp_table(name)` later